tests/PySys/monitoring/monitoring_with_collectd/run.py

```diff
@@ -0,0 +1,124 @@
+from pysys.basetest import BaseTest
+
+import time
+import re
+import json
+
+"""
+Validate tedge-dm-agent  messages that are published
+on tedge/measurements
+
+Given a configured system
+When we start the collectd with sudo in the background
+When we start the tedge-dm-agent with sudo in the background
+When we start tedge sub with sudo in the background
+Wait for couple of seconds to publish couple of batch of messages
+Then we kill tedge sub with sudo as it is running with a different user account
+Then we validate the  messages in the output of tedge sub,
+"""
+
+
+class MonitoringWithCollectd(BaseTest):
+
+    def setup(self):
+        self.js_msg = ""
+        self.cpu_cnt = 0
+        self.memory_cnt = 0
+        self.time_cnt = 0
+        self.disk_cnt = 0
+
+    def execute(self):
+        tedge = "/usr/bin/tedge"
+        sudo = "/usr/bin/sudo"
+
+        collectd = self.startProcess(
+            command=sudo,
+            arguments=["systemctl", "start", "collectd"],
+            stdouterr="collectd",
+        )
+
+        collectd_mapper = self.startProcess(
+            command=sudo,
+            arguments=["systemctl", "start", "tedge-dm-agent"],
+            stdouterr="collectd_mapper",
+        )
+
+        time.sleep(0.1)
+
+        sub = self.startProcess(
+            command=sudo,
+            arguments=[tedge, "mqtt", "sub", "--no-topic", "tedge/#"],
+            stdouterr="tedge_sub",
+            background=True,
+        )
+
+        # Wait for a small amount of time to give tedge sub time
+        # to initialize and capture couple of batches of messages
+        # that are published by tedge-dm-agent.
+        time.sleep(12)
+
+        # Kill the subscriber process explicitly with sudo as PySys does
+        # not have the rights to do it
+        kill = self.startProcess(
+            command=sudo,
+            arguments=["killall", "tedge"],
```

Comment:
That looks a bit risky as it's not just killing the process that we spawned but killing every instance of `tedge` running on the system.